src/main/kotlin/calculator/StringNumberCollection.kt

```diff
@@ -0,0 +1,11 @@
+package calculator
+
+class StringNumberCollection(private val numbers: List<StringNumber>) {
+    constructor(numbers: Array<String>) : this(numbers.map { StringNumber(it) })
+
+    fun sum(): Int {
+        return numbers.map {
+            it.toInt()
+        }.sum()
+    }
```

Comment:
```suggestion
    fun sum() = numbers.sumBy { it.toInt() }
```
이렇게하면 한줄로 표현이 가능합니다.